src/app/store/machine/slice.ts

```diff
@@ -1015,33 +1016,104 @@ const machineSlice = createSlice({
         // No state changes need to be handled for this action.
       },
     },
-    getError: (
-      state: MachineState,
-      action: PayloadAction<MachineState["errors"]>
-    ) => {
-      state.errors = action.payload;
-      state = setErrors(state, action, "get");
-      state.loading = false;
-      state.saving = false;
+    getError: {
+      prepare: (
+        item: { system_id: Machine[MachineMeta.PK] },
+        requestId: string,
+        errors: MachineStateDetailsItem["errors"]
+      ) => ({
+        meta: {
+          item,
+          requestId,
+        },
+        payload: errors,
+      }),
+      reducer: (
+        state: MachineState,
+        action: PayloadAction<
+          MachineStateDetailsItem["errors"],
+          string,
+          GenericItemMeta<{ system_id: Machine[MachineMeta.PK] }>
+        >
+      ) => {
+        if (action.meta.requestId && action.meta.requestId in state.details) {
+          state.details[action.meta.requestId].errors = action.payload;
+          state.details[action.meta.requestId].loading = false;
+        }
+        state = setErrors(state, action, "get");
+      },
     },
-    getStart: (state: MachineState) => {
-      state.loading = true;
+    getStart: {
+      prepare: (
+        item: { system_id: Machine[MachineMeta.PK] },
+        requestId: string
+      ) => ({
+        meta: {
+          item,
+          requestId,
+        },
+        payload: null,
+      }),
+      reducer: (
+        state: MachineState,
+        action: PayloadAction<
+          MachineStateDetailsItem["errors"],
+          string,
+          GenericItemMeta<{ system_id: Machine[MachineMeta.PK] }>
+        >
+      ) => {
+        if (action.meta.requestId) {
+          if (action.meta.requestId in state.details) {
+            state.details[action.meta.requestId].loading = true;
+          } else {
+            state.details[action.meta.requestId] = {
+              errors: null,
+              loaded: false,
+              loading: true,
+              system_id: action.meta.item.system_id,
+            };
+          }
+        }
+      },
     },
-    getSuccess: (state: MachineState, action: PayloadAction<Machine>) => {
-      const machine = action.payload;
-      // If the item already exists, update it, otherwise
-      // add it to the store.
-      const i = state.items.findIndex(
-        (draftItem: Machine) => draftItem.system_id === machine.system_id
-      );
-      if (i !== -1) {
-        state.items[i] = machine;
-      } else {
-        state.items.push(machine);
-        // Set up the statuses for this machine.
-        state.statuses[machine.system_id] = DEFAULT_STATUSES;
-      }
-      state.loading = false;
+    getSuccess: {
+      prepare: (
+        item: { system_id: Machine[MachineMeta.PK] },
+        requestId: string,
+        machine: MachineDetails
+      ) => ({
+        meta: {
+          item,
+          requestId,
```

Comment:
I think it's confusing that this `requestId` isn't the same as the `requestId` that already uniquely identifies each WebSocket message exchange.

Reply:
<img width="775" alt="image" src="https://user-images.githubusercontent.com/7452681/180491669-48df7533-982f-49be-bc4b-3717a9c11a0c.png">
<img width="323" alt="image" src="https://user-images.githubusercontent.com/7452681/180491732-7d8be50d-0a53-4c49-8baa-fc6b372fb188.png">

Reply:
Good point! I've stuck this in a new issue as it'll also affect another branch I have: https://github.com/canonical-web-and-design/app-tribe/issues/1226.